src/components/TodoList/TodoList.tsx

```diff
@@ -1,224 +1,100 @@
-              <p className="has-text-danger">molestiae perspiciatis ipsa</p>
-            </td>
-
-            <td className="has-text-right is-vcentered">
-              <button data-cy="selectButton" className="button" type="button">
-                <span className="icon">
-                  <i className="far fa-eye" />
-                </span>
-              </button>
-            </td>
-          </tr>
-
-          <tr data-cy="todo">
-            <td className="is-vcentered">10</td>
-
-            <td className="is-vcentered">
-              <span className="icon" data-cy="iconCompleted">
-                <i className="fas fa-check" />
-              </span>
-            </td>
-
-            <td className="is-vcentered is-expanded">
-              <p className="has-text-success">
-                illo est ratione doloremque quia maiores aut
-              </p>
-            </td>
-
-            <td className="has-text-right is-vcentered">
-              <button data-cy="selectButton" className="button" type="button">
+      {todos.length === 0 ? (
+        <p className="notification is-warning">
+          There are no todos matching current filter criteria
+        </p>
+      ) : (
+        <table className="table is-narrow is-fullwidth">
+          <thead>
+            <tr>
+              <th>#</th>
+
+              <th>
                 <span className="icon">
-                  <i className="far fa-eye" />
+                  <i className="fas fa-check" />
                 </span>
-              </button>
-            </td>
-          </tr>
-        </tbody>
-      </table>
+              </th>
+
+              <th>Title</th>
+              <th> </th>
+            </tr>
+          </thead>
+
+          <tbody>
+            {todos.map(todo => {
+              return (
+                <tr
+                  key={todo.id}
+                  data-cy="todo"
+                  className={classNames({
+                    'has-background-info-light':
+                      selectedTodo && selectedTodo.id === todo.id,
+                  })}
+                >
+                  <td className="is-vcentered">{todo.id}</td>
+                  <td className="is-vcentered">
+                    {todo.completed && (
+                      <span className="icon" data-cy="iconCompleted">
+                        <i className="fas fa-check" />
+                      </span>
+                    )}
+                  </td>
+                  <td className="is-vcentered is-expanded">
+                    <p
+                      className={
+                        todo.completed ? 'has-text-success' : 'has-text-danger'
+                      }
+                    >
+                      {todo.title}
+                    </p>
+                  </td>
+                  <td className="has-text-right is-vcentered">
+                    <button
+                      onClick={() => {
+                        dispatch(currentTodoActions.set(todo));
```

Comment:
Ensure that the `set` action from `currentTodoActions` is correctly defined and that `todo` contains all necessary properties expected by this action.